### PR TITLE
fix #701: Redirect user to the console when he tries to edit phone number on profile page

### LINKED ISSUE #701: Unable to modify phone number in profile
## Description

Two odd issues here. First is going into the users profile and tapping on the phone number makes the text smaller for no apparent reason. Second is editing the profile, it appears the number can be modified but accepts no input.

This also caused the app to freeze first I encountered it then unfroze itself sending me to the Chats view (I was tapping around, it maybe locked up and later registered a tap on the back arrow to exit). I was unable to reproduce this freezing issue. Maybe related to editing the phone field initially.

### PR DESCRIPTION
**Pull request by jumarko**
Since phone change requires sms verification, we don't want user let to edit it directly on profile page.
Instead, we redirect him to the Console automatically issuing :phone command.

### CODE REVIEW

File: src/status_im/profile/screen.cljs [hunk 1]
```diff
@@ -155,10 +163,13 @@
...
     [view st/profile-property-with-top-spacing
      [selectable-field {:label     (label :t/phone-number)
+
                         :editable? false
                         :value     (if (and phone (not (str/blank? phone)))
                                      (format-phone-number phone)
-                                     (label :t/not-specified))}]
+                                     (label :t/not-specified))
+                        ;; TODO: should this be changed?
+                        :on-press  navigate-to-phone-change}]
```
Comment (jumarko) [inline src/status_im/profile/screen.cljs]: Not sure if this should be changed - where is the profile view used?

File: src/status_im/profile/screen.cljs [hunk 2]
```diff
@@ -113,6 +114,13 @@
             [status-view {:style  (st/status-text (:height (r/state component)))
                           :status status}])])})))
 
+(defn- navigate-to-phone-change
+  "Switch user to the console issuing the !phone command automatically to let him change his phone number."
+  []
+  (dispatch [:navigate-to :chat console-chat-id])
+  (dispatch [:set-chat-command :phone])
+  )
```
Comment (jumarko) [inline src/status_im/profile/screen.cljs]: Is this acceptable or should I create new event like `:phone-number-change-requested` and dispatch these two events in the separate handler?

Comment (alwx) [inline src/status_im/profile/screen.cljs]: IMO this is acceptable, but it is better to move it to another handler just for clarity and readability

File: src/status_im/profile/screen.cljs (hunk 1 above)
Comment (alwx) [inline src/status_im/profile/screen.cljs]: The profile view you are talking about is another user's profile. You can access it by opening a chat with some user and pressing "Profile" in popup menu after that.
So the answer is no, we don't need to navigate to console chat in this case.

**Comment by alwx** [comment]
Thank you for your contribution!
We don't currently have unit tests, and I don't see any need for documentation changes.

**Comment by jumarko** [comment]
I removed the change of "other user profile" in fixup commit.

File: src/status_im/profile/screen.cljs [hunk 3]
```diff
@@ -155,10 +163,11 @@
 
     [view st/profile-property-with-top-spacing
      [selectable-field {:label     (label :t/phone-number)
+
                         :editable? false
                         :value     (if (and phone (not (str/blank? phone)))
                                      (format-phone-number phone)
-                                     (label :t/not-specified))}]
+                                     (label :t/not-specified))
```
Comment (rasom) [inline src/status_im/profile/screen.cljs]:
```
----  Could not Read  src/status_im/profile/screen.cljs   line:171  column:36  ----

  Unmatched delimiter ]

  169                                       (format-phone-number phone)
  170                                       (label :t/not-specified))
  171       [view st/underline-container]]
                                          ^--- Unmatched delimiter ]
  172
  173      (when address
  174        [view st/profile-property

----  Reader Error : Please see src/status_im/profile/screen.cljs  ----
```

**Comment by jumarko** [comment]
@rasom @alwx I've updated the pull request - it should compile now.
I also changed the approach a little bit by introducing new event :phone-number-change-requested which is dispatched when user clicks on phone field on profile screen.

**Comment by rasom** [comment]
Thanks for updates, @jumarko !

**Pull request merged**